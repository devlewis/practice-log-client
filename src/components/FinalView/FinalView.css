.finalview_p {
  font-size: 1em;
}

main {
  margin-bottom: 30px;
}
