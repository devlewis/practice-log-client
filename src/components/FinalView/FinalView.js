import React, { PureComponent } from "react";
import "./FinalView.css";
import { Link } from "react-router-dom";
import Context from "../../Context";

class FinalView extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {};
  }

  static contextType = Context;

  render() {
    const daysCompleted = this.context.days.filter(
      (day) => day.completed === "true"
    );
    const daysCompletedHours = this.context.days.filter(
      (day) => day.actual_hours >= this.context.hours_goal
    );

    const techniqueArr = [];
    const repertoireArr = [];

    this.context.days.forEach((day) => {
      if (day.technique !== "" && !techniqueArr.includes(day.technique)) {
        techniqueArr.push(day.technique);
      }
    });

    this.context.days.forEach((day) => {
      if (!repertoireArr.includes(day.repertoire) && day.repertoire !== "") {
        repertoireArr.push(day.repertoire);
      }
    });

    return (
      <main>
        <h1>YOU DID IT!</h1>
        <p className="finalview_p">
          You practiced {daysCompleted.length} out of {this.props.totalDays}{" "}
          goal days!
        </p>
        <p className="finalview_p">
          You reached your practice hourly goal {daysCompletedHours.length} out
          of {this.props.totalDays} days.
        </p>
        <p className="finalview_p">
          {" "}
          You practiced a total of {this.context.total_hours} hours during your{" "}
          {this.props.totalDays}-day goal.
        </p>
        <p className="finalview_p">Technique items recorded:</p>
        <ul className="technique_list">
          {techniqueArr.map((item) => {
            return <li>{item}</li>;
          })}
        </ul>
        <p className="finalview_p">Repertoire items recorded:</p>
        <ul className="repertoire_list">
          {repertoireArr.map((item) => (
            <li>{item}</li>
          ))}
        </ul>
        <Link to="/setup">
          <button>Set up a new goal!</button>
        </Link>
      </main>
    );
  }
}

export default FinalView;
