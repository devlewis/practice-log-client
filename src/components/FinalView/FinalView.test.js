import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import FinalView from "./FinalView";
import Context from "../../Context";
import { Route, BrowserRouter } from "react-router-dom";

let value = {
  days: [
    {
      id: 1,
      day_num: 1,
      date: "March 22",
      completed: "false",
      technique: "",
      repertoire: "",
      actual_hours: 0,
      touched: false,
      goal_id: 1,
      user_id: 1,
      goal_hours: 3,
    },
    {
      id: 2,
      day_num: 2,
      date: "March 23",
      completed: "true",
      technique: "Hanon",
      repertoire: "",
      actual_hours: 2,
      touched: true,
      goal_id: 1,
      user_id: 1,
      goal_hours: 3,
    },
  ],
  total_hours: 5,
  hours_goal: 3,
  num_of_days: 2,
};

describe(`FinalView component`, () => {
  const props = {
    history: {},
    totalDays: 2,
    className: "test-class-name",
    children: <p>test children</p>,
    "data-other": "test-other-prop",
  };

  it("renders a list FinalView given props", () => {
    const wrapper = mount(
      <BrowserRouter>
        <Route>
          <Context.Provider value={value}>
            <FinalView {...props} />
          </Context.Provider>
        </Route>
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
