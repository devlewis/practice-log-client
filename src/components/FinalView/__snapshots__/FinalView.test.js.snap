// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalView component renders a list FinalView given props 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Route>
      <FinalView
        className="test-class-name"
        data-other="test-other-prop"
        history={Object {}}
        totalDays={2}
      >
        <main>
          <h1>
            YOU DID IT!
          </h1>
          <p
            className="finalview_p"
          >
            You practiced 
            1
             out of 
            2
             
            goal days!
          </p>
          <p
            className="finalview_p"
          >
            You reached your practice hourly goal 
            0
             out of 
            2
             days.
          </p>
          <p
            className="finalview_p"
          >
             
            You practiced a total of 
            5
             hours during your
             
            2
            -day goal.
          </p>
          <p
            className="finalview_p"
          >
            Technique items recorded:
          </p>
          <ul
            className="technique_list"
          >
            <li
              key="0"
            >
              Hanon
            </li>
          </ul>
          <p
            className="finalview_p"
          >
            Repertoire items recorded:
          </p>
          <ul
            className="repertoire_list"
          />
          <Link
            to="/setup"
          >
            <LinkAnchor
              href="/setup"
              navigate={[Function]}
            >
              <a
                href="/setup"
                onClick={[Function]}
              >
                <button>
                  Set up a new goal!
                </button>
              </a>
            </LinkAnchor>
          </Link>
        </main>
      </FinalView>
    </Route>
  </Router>
</BrowserRouter>
`;
